### PR TITLE
Add Delete "X" Functionality + Cloudinary preparation added 

### PR DESCRIPTION
**Pull request by CTGitWhiz**
This pull request adds Delete "X" functionality to the project, allowing users to delete images from the image gallery. The feature is implemented using React's useState and useCallback hooks to handle deletion of images from the UI and local storage. Additionally, this pull request also prepares cloudinary integration which effectively stores and delivers optimized images for the app. Overall, these features enhance user experience in interacting with the application.

### CODE REVIEW

File: pages/components/ImageList.js [hunk 1]
```diff
@@ -27,34 +32,95 @@ const ImageListWrapper = styled.div`
...
+
+  //Set local storage using useMemo Hook
+  const setLocalStorage = useMemo(
+    () => () => {
+      if (typeof window !== "undefined") {
+        localStorage.setItem("images", JSON.stringify(imageList));
+      }
+    },
+    [imageList]
+  );
```
Comment (Kheirah) [inline pages/components/ImageList.js]: Are you using that hook anywhere?

File: pages/index.js [hunk 2]
```diff
@@ -18,11 +18,24 @@ const Title = styled.h1`
...
+export async function getServerSideProps() {
+  const images = Array.from({ length: 20 }, (_, i) => ({
+    src: `/image${i + 1}.png`,
+    alt: `My Image ${i + 1}`,
+    id: i + 1,
+  }));
+
+  // Returns props named images, which includes an array of 20 image objects
+  return { props: { images } };
+}
```
Comment (Kheirah) [inline pages/index.js:41]: Nice one. `getServerSideProps` is a special function in Next. Make sure to understand how it works.

**Review by Kheirah — APPROVED** [review]
Check the bit of code that's not being used and see if you need it.

File: pages/components/ImageList.js (hunk 1 above)
Comment (CTGitWhiz) [reply pages/components/ImageList.js]: Thanks for the notice :D 
It looks like it's just a remnant piece of code that I added at some point... and is no longer needed! --> Deleted!

**Pull request merged**